src/core/qgsvectorfilewriter.cpp

```diff
@@ -95,29 +93,29 @@ QgsVectorFileWriter::QgsVectorFileWriter(
         QString(), CreateOrOverwriteFile, newLayer, sinkFlags, transformContext, fieldNameSource );
 }
 
-QgsVectorFileWriter::QgsVectorFileWriter(
-  const QString &vectorFileName,
-  const QString &fileEncoding,
-  const QgsFields &fields,
-  Qgis::WkbType geometryType,
-  const QgsCoordinateReferenceSystem &srs,
-  const QString &driverName,
-  const QStringList &datasourceOptions,
-  const QStringList &layerOptions,
-  QString *newFilename,
-  Qgis::FeatureSymbologyExport symbologyExport,
-  FieldValueConverter *fieldValueConverter,
-  const QString &layerName,
-  ActionOnExistingFile action,
-  QString *newLayer,
-  const QgsCoordinateTransformContext &transformContext,
-  QgsFeatureSink::SinkFlags sinkFlags,
-  FieldNameSource fieldNameSource
-)
+QgsVectorFileWriter::QgsVectorFileWriter( const QString &vectorFileName,
```

Comment:
Surely this class has to win the highest-parameters-count-for-a-constructor award! ;-P

Reply:
I can't stand it...! Bring on QGIS 4.0 when we can rip this API up and streamline!